### PR TITLE
Fix bugs (string comparison) in csharp generators

### PR DESCRIPTION
**Pull request by wing328**
fix minor bugs in string comparison

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02) @Blackclaws (2021/03) @lucamazzanti (2021/05)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java [hunk 1]
```diff
@@ -504,7 +504,7 @@ public ModelsMap postProcessModels(ModelsMap objs) {
             if (cm.isEnum && !cm.vendorExtensions.containsKey(this.zeroBasedEnumVendorExtension)) {
                 if (Boolean.TRUE.equals(this.zeroBasedEnums)) {
                     cm.vendorExtensions.put(this.zeroBasedEnumVendorExtension, true);
-                } else if (!Boolean.FALSE.equals(this.zeroBasedEnums)) {
+                } else {
```
Comment (devhl-labs) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java:507]: I don't think we want this change. When the value is false, then the user has opted out. It should only go in here if the value is null.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java [hunk 2]
```diff
@@ -471,7 +471,7 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
         if (property.isEnum && !property.vendorExtensions.containsKey(this.zeroBasedEnumVendorExtension)) {
             if (Boolean.TRUE.equals(this.zeroBasedEnums)) {
                 property.vendorExtensions.put(this.zeroBasedEnumVendorExtension, true);
-            } else if (!Boolean.FALSE.equals(this.zeroBasedEnums)) {
+            } else {
```
Comment (devhl-labs) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java:474]: I don't think we want this change.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java:507]: thanks for the explanation. happy to revert the change.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java]: undo via 7df05a53b53

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java (hunk 2 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java]: undo via https://github.com/OpenAPITools/openapi-generator/commit/7df05a53b53bf605b52eadc37af4535acf5e8a70

**Pull request merged**